package ca.openosp.openo.commn.model.enumerator;

/**
 * Names of the per-provider settings stored in the property table.
 * Use these instead of the String constants in UserProperty.
 *
 * @since 2026-09-30
 */
public enum UserPropertyKey {

    STALE_NOTEDATE("cme_note_date"),
    STALE_FORMAT("cme_note_format"),
    ONTARIO_MD_USERNAME("ontario_md_username"),
    ONTARIO_MD_PASSWORD("ontario_md_password"),
    CONSULTATION_TIME_PERIOD_WARNING("consultation_time_period_warning"),
    CONSULTATION_TEAM_WARNING("consultation_team_warning"),
    WORKLOAD_MANAGEMENT("workload_management"),
    CONSULTATION_REQ_PASTE_FMT("consultation_req_paste_fmt"),
    CONSULTATION_LETTERHEADNAME_DEFAULT("consultation_letterheadname_default"),
    RX_PAGE_SIZE("rx_page_size"),
    RX_DEFAULT_QUANTITY("rx_default_quantity"),
    RX_PROFILE_VIEW("rx_profile_view"),
    RX_USE_RX3("rx_use_rx3"),
    DMFLOW_SHEET_VIEW("DMFlowsheet_view"),
    DOC_DEFAULT_QUEUE("doc_default_queue"),
    HC_TYPE("HC_Type"),
    DEFAULT_SEX("default_sex"),
    DEFAULT_REF_PRACTITIONER("default_ref_prac"),
    EFORM_REFER_FAX("eform_refer_fax"),
    EFORM_FAVOURITE_GROUP("favourite_eform_group"),
    RX_SHOW_PATIENT_DOB("rx_show_patient_dob"),
    PATIENT_NAME_LENGTH("patient_name_length"),

    OFFICIAL_FIRST_NAME("official_first_name"),
    OFFICIAL_SECOND_NAME("official_second_name"),
    OFFICIAL_LAST_NAME("official_last_name"),
    OFFICIAL_OLIS_IDTYPE("official_olis_idtype"),
    OSCAR_MSG_RECVD("oscarMsgRecvd"),
    LAB_MACRO_JSON("labMacroJSON"),

    //added to user properties with new interface
    FAX("fax"),
    SIGNATURE("signature"),
    COLOUR("colour"),
    SEX("sex"),
    SCHEDULE_START_HOUR("schedule.start_hour"),
    SCHEDULE_END_HOUR("schedule.end_hour"),
    SCHEDULE_PERIOD("schedule.period"),
    MYGROUP_NO("mygroup_no"),
    NEW_CME("new_cme"),
    ENCOUNTER_FORM_LENGTH("encounter.form_length"),
    ENCOUNTER_FORM_NAME("encounter.form_name"),
    EFORM_NAME("encounter.eform_name"),
    RX_SHOW_QR_CODE("rx_show_qr_code"),
    NEW_TICKLER_WARNING_WINDOW("new_tickler_warning_window"),
    CAISI_DEFAULT_PMM("caisi.default_pmm"),
    CAISI_PREV_BILLING("caisi.prev_billing"),
    DEFAULT_BILLING_FORM("default_billing_form"),
    DEFAULT_REFERRAL_TYPE("default_referral_type"),
    DEFAULT_PAYEE("default_payee"),
    DEFAULT_DX_CODE("default_dx_code"),
    CPP_SINGLE_LINE("cpp_single_line"),
    LAB_ACK_COMMENT("lab_ack_comment"),

    LAB_RECALL_DELEGATE("lab_recall_delegate"),
    LAB_RECALL_MSG_SUBJECT("lab_recall_msg_subject"),
    LAB_RECALL_TICKLER_ASSIGNEE("lab_recall_tickler_assignee"),
    LAB_RECALL_TICKLER_PRIORITY("lab_recall_tickler_priority"),

    EDOC_BROWSER_IN_MASTER_FILE("edoc_browser_in_master_file"),
    EDOC_BROWSER_IN_DOCUMENT_REPORT("edoc_browser_in_document_report"),
    VIEW_DOCUMENT_AS("view_document_as"),
    INCOMING_DOCUMENT_DEFAULT_QUEUE("incoming_document_default_queue"),
    INCOMING_DOCUMENT_ENTRY_MODE("incoming_document_entry_mode"),
    DISPLAY_DOCUMENT_AS("display_document_as"),
    PDF("PDF"),
    IMAGE("Image"),
    DOCUMENT_DESCRIPTION_TEMPLATE("document_description_template"),
    CLINIC("Clinic"),
    USER("User"),
    UPLOAD_DOCUMENT_DESTINATION("upload_document_destination"),
    INCOMINGDOCS("incomingDocs"),
    PENDINGDOCS("pendingDocs"),
    UPLOAD_INCOMING_DOCUMENT_FOLDER("upload_incoming_document_folder"),
    HIDE_OLD_ECHART_LINK_IN_APPT("hide_old_echart_link_in_appointment"),

    DEFAULT_PRINTER_PDF_LABEL("default_printer_pdf_label"),
    DEFAULT_PRINTER_PDF_ENVELOPE("default_printer_pdf_envelope"),
    DEFAULT_PRINTER_APPOINTMENT_RECEIPT("default_printer_appointment_receipt"),
    DEFAULT_PRINTER_PDF_ADDRESS_LABEL("default_printer_pdf_address_label"),
    DEFAULT_PRINTER_PDF_CHART_LABEL("default_printer_pdf_chart_label"),
    DEFAULT_PRINTER_CLIENT_LAB_LABEL("default_printer_client_lab_label"),
    DEFAULT_PRINTER_PDF_LABEL_SILENT_PRINT("default_printer_pdf_label_silent_print"),
    DEFAULT_PRINTER_PDF_ENVELOPE_SILENT_PRINT("default_printer_pdf_envelope_silent_print"),
    DEFAULT_PRINTER_APPOINTMENT_RECEIPT_SILENT_PRINT("default_printer_appointment_receipt_silent_print"),
    DEFAULT_PRINTER_PDF_ADDRESS_LABEL_SILENT_PRINT("default_printer_pdf_address_label_silent_print"),
    DEFAULT_PRINTER_PDF_CHART_LABEL_SILENT_PRINT("default_printer_pdf_chart_label_silent_print"),
    DEFAULT_PRINTER_CLIENT_LAB_LABEL_SILENT_PRINT("default_printer_client_lab_label_silent_print"),

    INTEGRATOR_DEMOGRAPHIC_SYNC("integrator_demographic_sync"),
    INTEGRATOR_DEMOGRAPHIC_ISSUES("integrator_demographic_issues"),
    INTEGRATOR_DEMOGRAPHIC_CONSENT("integrator_demographic_consent"),
    INTEGRATOR_DEMOGRAPHIC_ADMISSIONS("integrator_demographic_admissions"),
    INTEGRATOR_DEMOGRAPHIC_PREVENTIONS("integrator_demographic_preventions"),
    INTEGRATOR_DEMOGRAPHIC_NOTES("integrator_demographic_notes"),
    INTEGRATOR_DEMOGRAPHIC_DRUGS("integrator_demographic_drugs"),
    INTEGRATOR_DEMOGRAPHIC_APPOINTMENTS("integrator_demographic_appointments"),
    INTEGRATOR_DEMOGRAPHIC_DXRESEARCH("integrator_demographic_dxresearch"),
    INTEGRATOR_DEMOGRAPHIC_BILLING("integrator_demographic_billing"),
    INTEGRATOR_DEMOGRAPHIC_EFORMS("integrator_demographic_eforms"),
    INTEGRATOR_DEMOGRAPHIC_MEASUREMENTS("integrator_demographic_measurements"),
    INTEGRATOR_DEMOGRAPHIC_DOCUMENTS("integrator_demographic_documents"),
    INTEGRATOR_DEMOGRAPHIC_ALLERGIES("integrator_demographic_allergies"),
    INTEGRATOR_DEMOGRAPHIC_LABREQ("integrator_demographic_labreq"),
    INTEGRATOR_PROGRAMS("integrator_programs_sync"),
    INTEGRATOR_PROVIDERS("integrator_providers_sync"),
    INTEGRATOR_FACILITY("integrator_facility_sync"),
    INTEGRATOR_FULL_PUSH("integrator_full_push"),
    INTEGRATOR_LAST_PUSH("integrator_last_push"),
    INTEGRATOR_LAST_UPDATED("integrator_last_updated"),
    INTEGRATOR_LAST_PULL_PRIMARY_EMR("integrator_last_pull"),
    INTEGRATOR_PATIENT_CONSENT("integrator_patient_consent"),
    STUDENT_PARTICIPATION_CONSENT("student_participation_consent"),
    PROVIDER_FOR_TICKLER_WARNING("provider_for_tickler_warning"),

    MCEDT_ACCOUNT_PASSWORD("mcedt_account_password"),
    TICKLER_EMAIL_PROVIDER("tickler_email_provider"),

    DASHBOARD_SHARE("dashboard_share"),

    CODE_TO_ADD_PATIENTDX("code_to_add_patientDx"),
    CODE_TO_MATCH_PATIENTDX("code_to_match_patientDx"),

    PREVENTION_SSO_WARNING("prevention_sso_warning"),
    PREVENTION_ISPA_WARNING("prevention_ispa_warning"),
    PREVENTION_NON_ISPA_WARNING("prevention_non_ispa_warning"),

    TICKLER_TASK_ASSIGNEE("tickler_task_assignee"),

    EMAIL_COMMUNICATION("email_communication"),

    SCHEDULE_WEEK_VIEW_WEEKENDS("schedule.week_view_weekends"),
    RX_INTERACTION_WARNING_LEVEL("rxInteractionWarningLevel"),
    CONSULT_PASTE_HEADING("consult_paste_heading");

    private final String name;

    UserPropertyKey(String name) {
        this.name = name;
    }

    /**
     * Gets the name stored in the property table.
     *
     * @return String the property name, e.g. "schedule.week_view_weekends"
     */
    public String getName() {
        return this.name;
    }
}
